core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -284,6 +285,12 @@ private boolean readRequiredBytes() {
         totalBytesRead += toRead;
         nextFrame.addBuffer(unprocessed.readBytes(toRead));
       }
+      // There is no reliable way to get the uncompressed size per message when it's compressed,
+      // because the uncompressed bytes are provided through an InputStream whose total size is
+      // unknown until all bytes are read, and we don't know when it happens.
+      if (state == State.BODY) {
+        statsTraceCtx.inboundMessageRead(currentMessageSeqNo, requiredLength, -1);
```

Comment:
`processBody()` seems a better place to do this.

Reply:
Done.